### ISSUE TITLE
Mount partitions and skip HyperV limitations

### ISSUE THREAD

**Issue by murder** [microsoft/WSL#9283]
**Is your feature request related to a problem? Please describe.**
https://learn.microsoft.com/en-us/windows/wsl/wsl2-mount-disk#limitations

> At this time, only entire disks can be attached to WSL 2, meaning that it's not possible to attach only a partition. Concretely, this means that it's not possible to use wsl --mount to read a partition on the boot device, because that device can't be detached from Windows.

1. We cannot mount Linux partitions that share the disk with Windows main partitions
2. We cannot mount Linux partitions that share any Windows mounted partition
3. We cannot mount specific partitions - just whole disks

We cannot mount our own Linux partitions if we have them, if they share the same drive as Windows

**Describe the solution you'd like**

- Be able to mount my Linux partitions that share the disk with Windows
- Be able to mount other partitions without requiring to mount an entire disk
- Not needing to waste an isolated SSD/HDD for installing my native Linux distro solely to share with my WSL

**Describe alternatives you've considered**

- There are no alternatives

#8954 #6415 #4699

![DiskPart_output](https://user-images.githubusercontent.com/437087/206429329-902c2bca-bb05-4332-9aaa-93c5a936b2da.png)

![WSL_0x80070020](https://user-images.githubusercontent.com/437087/206429366-7347ceed-4930-4d43-9a75-734309978321.png)

**Comment by khdevnet** [comment]
It will be great to make possible mount ext4 partition what located on the same disk with windows boot system. On many laptops only one disk exist and no slot for one more disk, that block to use wsl2 on full power with ext4. Wait for that feature )))

**Comment by rbuels** [comment]
This would have been useful for me too.

**Comment by TapiaX** [comment]
In my case I opted to use the same disk (or drive cause is an ssd), most of us use partitioned disks so it is a shame not to have this feature

**Comment by alukaszew** [comment]
I never thought I'd say this but if nothing will change till the next week, I'm moving to linux + my mac. I have enough problems with windows in my work... 

Regards

**Comment by zgjja** [comment]
yes, this one is helpful. Managing VHD/VHDX in WSL2 is useful but not very compatible for users with both WSL2 and dual-system to access same files, so adding ability to support "--partition " option for real would be very convenient, also get better safety (VHD may sometimes get corrupted).

**Comment by ThiefMaster** [comment]
I'd also like to be able to mount an ext4 partition on the same disk that hosts the Windows system. This would be perfect to use a partition on that disk as a backup for my Linux system on another physical machine (by simply using rsync to make a copy).

**Comment by luohuiwu** [comment]
I met the same issue. I have only one disk and want to mount other ext4 partitions on the same disk that already hosted the Windows 10 OS.  I also agree that: it would be perfect to use a partition on that disk as a backup for my Linux system on another physical machine (by simply using rsync to make a copy).

**Comment by zccrs** [comment]
I need also this feature

**Comment by ifkato** [comment]
I need this too.

I want to access Linux partitions from WSL2 on a dual-boot laptop with only 1 SSD.

**Comment by yarohenning** [comment]
I just ran into the same issue (again). While trying to mount a LUKS encrypted partition.
```
$> wsl --mount \\.\PHYSICALDRIVE0 --partition 5 --bare
Failed to attach disk '\\.\PHYSICALDRIVE0' to WSL2: The process cannot access the file because it is being used by another process.
Error code: Wsl/Service/AttachDisk/MountDisk/ERROR_SHARING_VIOLATION
```
As long as I had a PC that had multiple disks switching between Linux and WSL and working on the same data worked like a charm. 
This is a huge limitation if people want to work on dual boot system where only one disk is possible.

Maybe @dkbennett, do you know who is the right one to include into this feature request, to provide info on when this is planned?

Thanks in advance!
